modules/liveintentRtdProvider.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * This module adds the liveintent provider to the Real Time Data module (rtdModule).
+ */
+import { submodule } from '../src/hook.js';
+import { logMessage } from '../src/utils.js';
+
+const SUBMODULE_NAME = 'liveintent';
+const GVLID = 148;
+
+/**
+ * Init
+ * @param {Object} config Module configuration
+ * @param {UserConsentData} userConsent User consent
+ * @returns true
+ */
+const init = (config, userConsent) => {
+  return true;
+}
+
+/**
+ * onBidRequest is called for each bidder during an auction and contains the bids for that bidder.
+ *
+ * @param {Object} bidRequest
+ * @param {SubmoduleConfig} config
+ * @param {UserConsentData} userConsent
+ */
+
+function onBidRequest(bidRequest, config, userConsent) {
+  bidRequest.bids.forEach(bid => {
+    const userIdSegments = { segment: bid.userId.lipbid.segments.map(id => ({ id })) }
+    const liSegments = [{name: 'liveintent.com', ...userIdSegments}]
+    if (bidRequest?.ortb2?.user?.data) {
+      bidRequest.ortb2.user.data = bidRequest.ortb2.user.data.concat(liSegments)
```

Comment:
Currently, you iterate over `bids`, but ignore them and keep concatenating the same data multiple times. 

The goal is this: Enrich each `bid` (`bid.ortb2.user.data`) according to the `segments` known to that `bid`. Additionally enrich the `bidRequest` (bidRequest.ortb2.user.data) according to the `segments` known to it.

Reply:
Thanks.. I just saw that each bid has ortb2.... I confused it with the object with the ortb2 in the root ..

Reply:
I fixed the logic and tested it.